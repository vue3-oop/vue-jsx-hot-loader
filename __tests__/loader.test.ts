import compiler from './compiler';
import { resolve } from 'path';

describe('vue-jsx-hot-loader test cases', () => {

  it('should work with export default in jsx', async () => {
    const A = await compiler(resolve(__dirname, '../example/A.jsx'));
    expect(A).toMatchSnapshot();
  });

  it('should work with export default referencing variable declaration', async () => {
    const App = await compiler(resolve(__dirname, '../example/App.jsx'));
    expect(App).toMatchSnapshot();
  });

  it('should work with named exports', async () => {
    const B = await compiler(resolve(__dirname, '../example/B.jsx'));
    expect(B).toMatchSnapshot();
  });

  it('should work with export default in tsx', async () => {
    const C = await compiler(resolve(__dirname, '../example/D.tsx'));
    expect(C).toMatchSnapshot();
  });
});
