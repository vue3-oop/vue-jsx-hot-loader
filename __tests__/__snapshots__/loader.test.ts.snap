// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`vue-jsx-hot-loader test cases file A should be transformed right 1`] = `
"import { createVNode as _createVNode, createTextVNode as _createTextVNode } from \\"vue\\";
export default (() => _createVNode(\\"div\\", null, [_createTextVNode(\\"23411dd\\")]));"
`;

exports[`vue-jsx-hot-loader test cases file App should be transformed right 1`] = `
"import { createVNode as _createVNode, createTextVNode as _createTextVNode, Fragment as _Fragment } from \\"vue\\";
import { defineComponent } from \\"vue\\";
import A from \\"./A\\";
import { B } from \\"./B\\";
import C from \\"./C\\";
const App = defineComponent({
  data() {
    return {
      a: 1
    };
  },

  render() {
    const {
      a
    } = this;
    return _createVNode(_Fragment, null, [a, _createVNode(\\"div\\", {
      \\"onClick\\": () => {
        this.a++;
      }
    }, [_createTextVNode(\\"Hello World!\\")]), _createVNode(A, null, null), _createVNode(B, null, null), _createVNode(C, null, null)]);
  }

});
export default App;
App.__hmrId = '1e8b21a4';

__VUE_HMR_RUNTIME__.createRecord('1e8b21a4', App);
/* hot reload */


if (module.hot) {
  module.hot.accept();

  __VUE_HMR_RUNTIME__.reload(\\"1e8b21a4\\", App);
}"
`;

exports[`vue-jsx-hot-loader test cases file B should be transformed right 1`] = `
"import { createVNode as _createVNode, createTextVNode as _createTextVNode, Fragment as _Fragment } from \\"vue\\";
import { defineComponent } from \\"vue\\";
const B = defineComponent({
  data() {
    return {
      a: 1
    };
  },

  render() {
    const {
      a
    } = this;
    return _createVNode(_Fragment, null, [_createVNode(\\"div\\", {
      \\"onClick\\": () => {
        this.a++;
      }
    }, [a, _createTextVNode(\\"d4s\\")]), _createVNode(\\"span\\", null, [_createTextVNode(\\"2\\")])]);
  }

});
export { B };
B.__hmrId = '9c2d2320';

__VUE_HMR_RUNTIME__.createRecord('9c2d2320', B);
/* hot reload */


if (module.hot) {
  module.hot.accept();

  __VUE_HMR_RUNTIME__.reload(\\"9c2d2320\\", B);
}"
`;

exports[`vue-jsx-hot-loader test cases file C should be transformed right 1`] = `
"import { createVNode as _createVNode } from \\"vue\\";
import { defineComponent } from \\"vue\\";

const __default__ = defineComponent({
  setup() {
    const string = \\"component C\\";
    return () => _createVNode(\\"div\\", null, [string]);
  }

});

export default __default__;
__default__.__hmrId = '69836bfa';

__VUE_HMR_RUNTIME__.createRecord('69836bfa', __default__);
/* hot reload */


if (module.hot) {
  module.hot.accept();

  __VUE_HMR_RUNTIME__.reload(\\"69836bfa\\", __default__);
}"
`;

exports[`vue-jsx-hot-loader test cases file t should be transformed right 1`] = `"export const test = \\"123\\";"`;

exports[`vue-jsx-hot-loader test cases should work with export default in jsx 1`] = `
"import { createVNode as _createVNode, createTextVNode as _createTextVNode } from \\"vue\\";
export default (() => _createVNode(\\"div\\", null, [_createTextVNode(\\"23411dd\\")]));"
`;

exports[`vue-jsx-hot-loader test cases should work with export default in tsx 1`] = `
"import { createVNode as _createVNode } from \\"vue\\";
import { defineComponent } from \\"vue\\";

const __default__ = defineComponent({
  setup() {
    const string = \\"component C\\";
    return () => _createVNode(\\"div\\", null, [string]);
  }

});

export default __default__;
__default__.__hmrId = 'b7ebfcf0';

__VUE_HMR_RUNTIME__.createRecord('b7ebfcf0', __default__);
/* hot reload */


if (module.hot) {
  module.hot.accept();

  __VUE_HMR_RUNTIME__.reload(\\"b7ebfcf0\\", __default__);
}"
`;

exports[`vue-jsx-hot-loader test cases should work with export default referencing variable declaration 1`] = `
"import { createVNode as _createVNode, createTextVNode as _createTextVNode, Fragment as _Fragment } from \\"vue\\";
import { defineComponent } from 'vue';
import A from './A';
import { B } from './B';
import C from './C';
const App = defineComponent({
  data() {
    return {
      a: 1
    };
  },

  render() {
    const {
      a
    } = this;
    return _createVNode(_Fragment, null, [a, _createVNode(\\"div\\", {
      \\"onClick\\": () => {
        this.a++;
      }
    }, [_createTextVNode(\\"Hello World!\\")]), _createVNode(A, null, null), _createVNode(B, null, null), _createVNode(C, null, null)]);
  }

});
export default App;
App.__hmrId = '4a166efb';

__VUE_HMR_RUNTIME__.createRecord('4a166efb', App);
/* hot reload */


if (module.hot) {
  module.hot.accept();

  __VUE_HMR_RUNTIME__.reload(\\"4a166efb\\", App);
}"
`;

exports[`vue-jsx-hot-loader test cases should work with named exports 1`] = `
"import { createVNode as _createVNode, createTextVNode as _createTextVNode, Fragment as _Fragment } from \\"vue\\";
import { defineComponent } from 'vue';
const B = defineComponent({
  data() {
    return {
      a: 1
    };
  },

  render() {
    const {
      a
    } = this;
    return _createVNode(_Fragment, null, [_createVNode(\\"div\\", {
      \\"onClick\\": () => {
        this.a++;
      }
    }, [a, _createTextVNode(\\"d4s\\")]), _createVNode(\\"span\\", null, [_createTextVNode(\\"2\\")])]);
  }

});
export { B };
B.__hmrId = '6f548b3d';

__VUE_HMR_RUNTIME__.createRecord('6f548b3d', B);
/* hot reload */


if (module.hot) {
  module.hot.accept();

  __VUE_HMR_RUNTIME__.reload(\\"6f548b3d\\", B);
}"
`;
